src/Router.js

```diff
@@ -44,7 +44,7 @@ export default class Router extends React.Component {
   componentDidMount() {
     if (!this.props.dispatch) {
       navigationStore.dispatch = this.navigator._navigation.dispatch;
-      navigationStore.getState = () => this.navigator._navigation.state;
+      navigationStore.getState = () => this.navigator.state.nav;
 
```

Comment:
I still don't understand why we can't use own state? This way react-navigation should not maintain own one (as their document says).

Reply:
The problem also that main idea of mobx integration was to make `currentScene` observable, i.e. re-render screens that use `currentScene`. It is usable for example to display `loading` buttons if `prevScene != currentScene`. I will check how it works with mobx-less implementation...

Reply:
I am trying to avoid having our own state because it brings little benefit to try and maintain it ourselves vs letting `react-navigation` handle it. 

Without `MobX` or something of that nature I don't think you will be able to easily cause a re-render when `prevScene` or `currentScene` change.